interfaces/cython/cantera/thermo.pyi

```diff
@@ -0,0 +1,604 @@
+        self,
+        fuel: CompositionLike,
+        oxidizer: CompositionLike,
+        basis: Basis = "mole",
+        element: str | bytes | float = "Bilger",
+    ) -> float: ...
+    def stoich_air_fuel_ratio(
+        self,
+        fuel: CompositionLike,
+        oxidizer: CompositionLike,
+        basis: Basis = "mole",
+    ) -> float: ...
+    def elemental_mass_fraction(self, m: str | bytes | float) -> float: ...
+    def elemental_mole_fraction(self, m: str | bytes | float) -> float: ...
+    def set_unnormalized_mass_fractions(self, Y: ArrayLike) -> None: ...
+    def set_unnormalized_mole_fractions(self, X: ArrayLike) -> None: ...
+    def mass_fraction_dict(self, threshold: float = 0.0) -> dict[str, float]: ...
+    def mole_fraction_dict(self, threshold: float = 0.0) -> dict[str, float]: ...
+    ######## Read-only thermodynamic properties ########
+    @property
+    def P(self) -> float: ...
+    @property
+    def T(self) -> float: ...
+    @property
+    def density(self) -> float: ...
+    @property
+    def density_mass(self) -> float: ...
+    @property
+    def density_mole(self) -> float: ...
+    @property
+    def v(self) -> float: ...
+    @property
+    def volume_mass(self) -> float: ...
+    @property
+    def volume_mole(self) -> float: ...
+    @property
+    def u(self) -> float: ...
+    @property
+    def int_energy_mole(self) -> float: ...
+    @property
+    def int_energy_mass(self) -> float: ...
+    @property
+    def h(self) -> float: ...
+    @property
+    def enthalpy_mole(self) -> float: ...
+    @property
+    def enthalpy_mass(self) -> float: ...
+    @property
+    def s(self) -> float: ...
+    @property
+    def entropy_mole(self) -> float: ...
+    @property
+    def entropy_mass(self) -> float: ...
+    @property
+    def g(self) -> float: ...
+    @property
+    def gibbs_mole(self) -> float: ...
+    @property
+    def gibbs_mass(self) -> float: ...
+    @property
+    def cv(self) -> float: ...
+    @property
+    def cv_mole(self) -> float: ...
+    @property
+    def cv_mass(self) -> float: ...
+    @property
+    def cp(self) -> float: ...
+    @property
+    def cp_mole(self) -> float: ...
+    @property
+    def cp_mass(self) -> float: ...
+    @property
+    def critical_temperature(self) -> float: ...
+    @property
+    def critical_pressure(self) -> float: ...
+    @property
+    def critical_density(self) -> float: ...
+    @property
+    def P_sat(self) -> float: ...
+    @property
+    def T_sat(self) -> float: ...
+    @property
+    def auxiliary_data(self) -> dict[str, Any]: ...
+    ######## Methods to get/set the complete thermodynamic state ########
+    @property
+    def state_size(self) -> int: ...
+    @property
+    def state(self) -> Array: ...
+    @state.setter
+    def state(self, state: ArrayLike) -> None: ...
+    @property
+    def TD(self) -> tuple[float, float]: ...
+    @TD.setter
+    def TD(self, values: Sequence[float]) -> None: ...
```

Comment:
Like we discussed for the `SolutionArray` and `Quantity` classes, I think these should also be restricted to `tuple[float, float]`. Same for the setters of the `PureFluid` class.